cmd/buildah/images.go

```diff
@@ -270,48 +272,52 @@ func outputImages(ctx context.Context, images []storage.Image, store storage.Sto
 			names = image.Names
 		} else {
 			// images without names should be printed with "<none>" as the image name
-			names = append(names, "<none>")
+			names = append(names, "<none>:<none>")
 		}
-		for _, name := range names {
-			if !matchesReference(name, argName) {
-				continue
-			}
-			found = true
+		for name, tags := range imagebuildah.ReposToMap(names) {
+			for _, tag := range tags {
 
-			if !matchesFilter(ctx, image, store, name, filters) {
-				continue
-			}
-			if opts.quiet {
-				fmt.Printf("%-64s\n", image.ID)
-				// We only want to print each id once
-				break
-			}
+				if !matchesReference(name, argName) {
+					continue
+				}
+				found = true
 
-			if opts.json {
-				JSONImage := jsonImage{ID: image.ID, Names: image.Names}
-				data, err2 := json.MarshalIndent(JSONImage, "", "    ")
-				if err2 != nil {
-					return err2
+				if !matchesFilter(ctx, image, store, name, filters) {
+					continue
+				}
+				if opts.quiet {
+					fmt.Printf("%-64s\n", image.ID)
+					// We only want to print each id once
+					break
```

Comment:
This break is breaking out of the new `for _, tag := range tags {` loop now, but we need to break out of the original for loop `for name, tags := range imagebuildah.ReposToMap(names) {` still.  Need to figure out how to break and go to the next name and stop processing on the tags when --quiet is selected.